src/py4vasp/scripts/ErrorAnalysis.py

```diff
@@ -0,0 +1,603 @@
+        print("Writing the energy error in meV/atom to file ", fname)
+        print("the format is structure index vs error")
+        print("~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~")
+        data = self.prepare_output_array(self.xaxis, self.energy_error[:, 0])
+        with open(fname, "w") as outfile:
+            outfile.write(
+                "# RMSE energy/atom in meV/atom "
+                + self.format_float([self.rmse_energy])
+                + "\n"
+            )
+            outfile.write(
+                "#      structure index "
+                + "  energy difference in meV/atom "
+                + "(value > 0 MLFF predicts too high value)"
+                + "\n"
+            )
+            for i in range(data.shape[0]):
+                outfile.write(self.format_float(data[i, :]) + "\n")
+
+    def writing_force_error_file(self, fname="ForceError.out"):
+        """
+        writing the force error to a output file
+        the output file format will be
+            structure index | root mean square force error in meV/Angstroem
+        """
+        print("~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~")
+        print("Writing the force error in meV/Angstroem per atom to file ", fname)
+        print("the format is structure index vs error")
+        print("~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~")
+        data = self.prepare_output_array(self.xaxis, self.force_error[:, 0])
+        with open(fname, "w") as outfile:
+            outfile.write(
+                "# RMSE force in meV/Angstroem "
+                + self.format_float([self.rmse_force])
+                + "\n"
+            )
+            outfile.write(
+                "#       structure index " + "  RMSE in meV/Angstroem " + "\n"
+            )
+            for i in range(data.shape[0]):
+                outfile.write(self.format_float(data[i, :]) + "\n")
+
+    def writing_stress_error_file(self, fname="StressError.out"):
+        """
+        writing the stress error to a output file
+        the output file format will be
+            structure index | root mean square stress error in [kBar]
+        """
+        print("~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~")
+        print("Writing the stress error in [kBar] to file ", fname)
+        print("the format is structure index vs error")
+        print("~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~")
+
+        data = self.prepare_output_array(self.xaxis, self.stress_error[:, 0])
+        with open(fname, "w") as outfile:
+            outfile.write(
+                "# RMSE stress in kilo-bar "
+                + self.format_float([self.rmse_stress])
+                + "\n"
+            )
+            outfile.write("#      structure index " + "     RMSE in kbar " + "\n")
+            for i in range(data.shape[0]):
+                outfile.write(self.format_float(data[i, :]) + "\n")
```

Comment:
Perhaps writing in a standard format would be more convenient? Personally, I would probably try csv and write all quantities into a single file
~~~python
import pandas as pd
data = {"energy error": self.energy_error[:,0], "force error": self.force_error[:,0], "stress error": self.stress_error[:,0]}
dataframe = pd.DataFrame(data=data)
dataframe.to_csv("error_analysis.csv")
~~~